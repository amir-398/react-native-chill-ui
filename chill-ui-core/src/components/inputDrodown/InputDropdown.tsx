import DropdownBase from './InputDropdownBase';
import DropdownList from './InputDropdownList';
import { InputDropdownProps } from '../../types/dropdown.types';

export default function InputDropdown(props: InputDropdownProps) {
  const {
    className,
    data,
    dropdownItemProps,
    DropdownItemRender,
    dropdownListProps,
    emptyText,
    hasAnimation,
    hasSearch,
    hasShadow,
    isLoading,
    itemClickableAs,
    maxHeight,
    minHeight,
    onSelectItem,
    searchInputProps,
    visible,
  } = props;

  return (
    <DropdownBase
      visible={visible}
      maxHeight={maxHeight}
      minHeight={minHeight}
      className={className}
      hasShadow={hasShadow}
      hasAnimation={hasAnimation}
      hasSearch={hasSearch}
      searchInputProps={searchInputProps}
    >
      <DropdownList
        data={data}
        onSelectItem={onSelectItem}
        dropdownItemProps={dropdownItemProps}
        DropdownItemRender={DropdownItemRender}
        dropdownListProps={dropdownListProps}
        isLoading={isLoading}
        emptyText={emptyText}
        itemClickableAs={itemClickableAs}
      />
    </DropdownBase>
  );
}
