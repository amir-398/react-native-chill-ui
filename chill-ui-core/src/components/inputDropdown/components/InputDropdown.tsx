import { InputDropdownProps } from '@types';
import { classNamePropsHandler } from '@utils';

import DropdownBase from './InputDropdownBase';
import DropdownList from './InputDropdownList';

/**

 * Provides a comprehensive dropdown list with search functionality, loading states, and customizable styling.
 * Automatically detects NativeWind availability and falls back to StyleSheet if needed.
 *
 * @example
 * ```tsx
 * <InputDropdown
 *   visible={isOpen}
 *   data={options}
 *   hasSearch
 *   searchInputProps={{ placeholder: "Search options..." }}
 *   isLoading={loading}
 *   onSelectItem={(item) => setSelectedItem(item)}
 *   maxHeight={400}
 *   className="bg-white rounded-lg shadow-lg"
 * />
 * ```
 *
 * @param visible - Whether the dropdown is visible and should be rendered
 * @param data - Array of data items to display in the dropdown list
 * @param maxHeight - Maximum height of the dropdown container (required)
 * @param minHeight - Minimum height of the dropdown container
 * @param onSelectItem - Callback function when an item is selected from the list
 * @param className - Custom CSS classes for the dropdown container (Tailwind/CSS)
 * @param style - Custom style object for the dropdown container
 * @param hasShadow - Whether to apply shadow styling to the dropdown
 * @param hasSearch - Whether to show a search input at the top of the dropdown
 * @param hasAnimation - Whether to enable opening/closing animations (default: true)
 * @param searchInputProps - Props to pass to the search Input component
 * @param customSearchInput - Custom search input component to replace the default
 * @param emptyText - Text to display when the data array is empty
 * @param isLoading - Whether the dropdown is in a loading state
 * @param customEmpty - Custom component to render when the list is empty
 * @param customLoadingIndicator - Custom loading indicator component
 * @param dropdownItemProps - Props to pass to each dropdown item component
 * @param loadingIndicatorProps - Props for the default loading indicator
 * @param DropdownItemRender - Custom render function for dropdown items
 * @param dropdownListProps - Props to pass to the underlying FlatList component
 * @param itemClickableAs - Type of touchable component for items ('TouchableOpacity' | 'Pressable' | 'TouchableHighlight' | 'RipplePressable' | 'none')
 * @returns Styled dropdown component with search, loading states, and item selection
 */
export default function InputDropdown(props: InputDropdownProps) {
  classNamePropsHandler(props, 'InputDropdown');
  const {
    className,
    data,
    dropdownItemProps,
    DropdownItemRender,
    dropdownListProps,
    emptyText,
    hasAnimation,
    hasSearch,
    hasShadow,
    isLoading,
    itemClickableAs,
    maxHeight,
    minHeight,
    onSelectItem,
    searchInputProps,
    style,
    visible,
  } = props;

  return (
    <DropdownBase
      visible={visible}
      maxHeight={maxHeight}
      minHeight={minHeight}
      className={className}
      hasShadow={hasShadow}
      style={style}
      hasAnimation={hasAnimation}
      hasSearch={hasSearch}
      searchInputProps={searchInputProps}
    >
      <DropdownList
        data={data}
        onSelectItem={onSelectItem}
        dropdownItemProps={dropdownItemProps}
        DropdownItemRender={DropdownItemRender}
        dropdownListProps={dropdownListProps}
        isLoading={isLoading}
        emptyText={emptyText}
        itemClickableAs={itemClickableAs}
      />
    </DropdownBase>
  );
}
