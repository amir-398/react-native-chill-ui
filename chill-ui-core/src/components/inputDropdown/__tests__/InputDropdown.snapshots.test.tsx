import { render } from '@testing-library/react-native';

import InputDropdown from '../components/InputDropdown';

// Mocks simples
jest.mock('../styles/InputDropdownBase.ss.styles', () => ({
  styles: {},
}));

jest.mock('../../../utils', () => ({
  classNameHandler: jest.fn(() => ({})),
  classNamePropsHandler: jest.fn(),
  cn: jest.fn((...args) => args.filter(Boolean).join(' ')),
  styleHandler: jest.fn(() => ({})),
}));

jest.mock('../../../components/box', () => ({
  Box: ({ children }: any) => children,
}));

jest.mock('../../../components/input', () => ({
  Input: ({ children }: any) => <div>{children}</div>,
}));

jest.mock('../../../components/animatedBox', () => ({
  AnimatedBox: ({ children }: any) => children,
}));

jest.mock('../components/InputDropdownBase', () => ({
  __esModule: true,
  default: ({ children }: any) => children,
}));

jest.mock('../components/InputDropdownList', () => ({
  __esModule: true,
  default: ({ data }: any) => (
    <div>
      {data?.map((item: any, index: number) => (
        <div key={index}>{item.label || item.name || item}</div>
      ))}
    </div>
  ),
}));

describe('InputDropdown Component Snapshots', () => {
  const mockData = [
    { id: 1, label: 'Option 1' },
    { id: 2, label: 'Option 2' },
    { id: 3, label: 'Option 3' },
  ];

  it('should match snapshot for basic dropdown', () => {
    const tree = render(<InputDropdown visible data={mockData} onSelectItem={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with search', () => {
    const tree = render(
      <InputDropdown
        visible
        data={mockData}
        hasSearch
        searchInputProps={{ placeholder: 'Search options' }}
        onSelectItem={jest.fn()}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with loading', () => {
    const tree = render(<InputDropdown visible data={[]} isLoading onSelectItem={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with empty state', () => {
    const tree = render(<InputDropdown visible data={[]} emptyText="No options available" onSelectItem={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot for dropdown with all props', () => {
    const tree = render(
      <InputDropdown
        visible
        data={mockData}
        hasSearch
        hasAnimation
        hasShadow
        maxHeight={300}
        minHeight={50}
        className="custom-dropdown"
        emptyText="No items"
        onSelectItem={jest.fn()}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot for hidden dropdown', () => {
    const tree = render(<InputDropdown visible={false} data={mockData} onSelectItem={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
