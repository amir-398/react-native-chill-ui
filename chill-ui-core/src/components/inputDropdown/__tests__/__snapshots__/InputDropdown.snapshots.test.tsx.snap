// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDropdown Component Snapshots should match snapshot for basic dropdown 1`] = `
<div>
  <div>
    Option 1
  </div>
  <div>
    Option 2
  </div>
  <div>
    Option 3
  </div>
</div>
`;

exports[`InputDropdown Component Snapshots should match snapshot for dropdown with all props 1`] = `
<div>
  <div>
    Option 1
  </div>
  <div>
    Option 2
  </div>
  <div>
    Option 3
  </div>
</div>
`;

exports[`InputDropdown Component Snapshots should match snapshot for dropdown with empty state 1`] = `<div />`;

exports[`InputDropdown Component Snapshots should match snapshot for dropdown with loading 1`] = `<div />`;

exports[`InputDropdown Component Snapshots should match snapshot for dropdown with search 1`] = `
<div>
  <div>
    Option 1
  </div>
  <div>
    Option 2
  </div>
  <div>
    Option 3
  </div>
</div>
`;

exports[`InputDropdown Component Snapshots should match snapshot for hidden dropdown 1`] = `
<div>
  <div>
    Option 1
  </div>
  <div>
    Option 2
  </div>
  <div>
    Option 3
  </div>
</div>
`;
